### DISCUSSION TITLE
Autocomplete insert vs. overwrite

### DISCUSSION THREAD

**Discussion by bazhenov** [zed-industries/zed#6482] [Feature Requests]
First of all, thank you very much for your work. I must say zed is one of the most pleasant experiences I have in coding for last several years. It's freaking fast, and I very like editor is not to staying in my way. Great job!

Right now (zed 0.76.1) when press <kbd>Enter</kbd> or <kbd>Tab</kbd> in autocomplete, selected item overwrites the item next to it (if there is no whitespace to the right of the cursor). In previous zed version item was appended. I think it would be useful to have both strategies available.

For example in IntelliJ IDEA <kbd>Enter</kbd> is inserting text <kbd>Tab</kbd> is overwriting.

**Comment by tjstarak** [comment]
Not sure if an option to configure that has already been implemented, but just want to highlight that the default behavior is frustrating.  I haven't found any information online on how to change it.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
